plugins/modules/foreman_external_usergroup.py

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2019 Kirill Shirinkin (kirill@mkdev.me)
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: foreman_external_usergroup
+short_description: Manage Foreman User groups
+description:
+  - Create and delete external user groups in Foreman
+author:
+  - "Kirill Shirinkin (@Fodoj)"
+options:
+  name:
+    description:
+      - Name of the group
+    required: true
+    type: str
+  usergroup:
+    description:
+      - Name of the linked usergroup
+    required: true
+    type: str
+  auth_source:
+    description:
+      - Name of the authentication source to be used for this group
+    required: true
+    type: str
+  state:
+    description:
+      - State of the user
+    default: present
+    choices:
+      - present
+      - absent
+    type: str
+extends_documentation_fragment: foreman
+'''
+
+EXAMPLES = '''
+- name: Create an external user group
+  foreman_external_usergroup:
+    name: test
+    auth_source: "My LDAP server"
+    usergroup: "Internal Usergroup"
+    state: present
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
+
+
+def main():
+    module = ForemanEntityAnsibleModule(
+        entity_spec=dict(
+            name=dict(required=True),
+            usergroup=dict(required=True, type='str'),
+            auth_source=dict(required=True, type='entity', flat_name='auth_source_id'),
+        ),
+    )
+
+    entity_dict = module.clean_params()
+
+    module.connect()
+
+    params = {"usergroup_id": entity_dict.pop('usergroup')}
```

Comment:
This is interesting, as the param name is `usergroup_id`, but the API actually also allows the name attribute here.

Reply:
What I wanted to say here: I'm debating with myself whether I like this, or want to see a `find_resource_by_name` just to be consistent.

@bagasse @mdellweg opinions?

Reply:
Well, saves one API call if we just use the name :-)

Reply:
I know...

The thing is, I think it's a miss-feature of the API to accept the name here too (but it does this in a few places) and a part of me says "the field is called `_id`, so put in a damn id there".

And on the other hand, it saves a request. :woman_shrugging:

Reply:
My guess is foreman uses friendly_id ruby on rails gem to treat some fields same as IDs. IMO if actual backend API  thinks it's okay to accept name as ID and it's described in API spec then API clients can do so as well. :)

Reply:
Yeah, Foreman does:
https://github.com/theforeman/foreman/blob/cd8d94440318e8ed4f8a943df5142b776bba5d62/app/models/external_usergroup.rb#L2-L3

I'm fine with this as it is I think.

Reply:
Fine for me too.

Reply:
@evgeni for me it's ok, as it is documented in the API doc that it accept `ID or name of user group`, and :cherries:  on the cake it save 1 API call, so :+1: